rllib/core/learner/torch/torch_meta_learner.py

```diff
@@ -0,0 +1,453 @@
+    override,
+    OverrideToImplementCustomLogic,
+    OverrideToImplementCustomLogic_CallToSuperRecommended,
+)
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.metrics import (
+    DATASET_NUM_ITERS_TRAINED,
+    DATASET_NUM_ITERS_TRAINED_LIFETIME,
+    DIFFERENTIABLE_LEARNER_RESULTS,
+)
+from ray.rllib.utils.metrics.utils import to_snake_case
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.typing import (
+    EpisodeType,
+    ModuleID,
+    NamedParamDict,
+    ResultDict,
+    TensorType,
+)
+
+logger = logging.getLogger("__name__")
+
+torch, nn = try_import_torch()
+
+
+class TorchMetaLearner(TorchLearner):
+    """A `TorchLearner` designed for meta-learning with functional updates.
+
+    This `TorchLearner` manages one or more `DifferentiableLearner` instances,
+    which perform functional updates on the `MultiRLModule`. These updates enable
+    the computation of higher-order gradients, making this class suitable for
+    meta-learning applications.
+
+    The `update` method executes one or more update loops on its
+    `DifferentiableLearner` instances, leveraging the functionally updated
+    parameters for its own learning process.
+    """
+
+    # A list of `TorchDifferentiableLearner`s that run inner functional update
+    # loops.
+    others: List[TorchDifferentiableLearner]
+
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        # First, initialize the `TorchLearner`.
+        super().__init__(**kwargs)
+
+        # Initialize all configured `TorchDifferentiableLearner`s.
+        self.others = [
+            other_config.learner_class(
+                config=self.config,
+                learner_config=other_config,
+                module=self.module,
+            )
+            for other_config in self.config.differentiable_learner_configs
+        ]
+
+    @OverrideToImplementCustomLogic_CallToSuperRecommended
+    @override(TorchLearner)
+    def build(self) -> None:
+        """Builds the `TorchMetaLearner`.
+
+        This method should be called before the meta-learner is used. It is
+        responsible for setting up the `TorchDifferentiableLearner`s.
+        """
+        super().build()
+
+        # Build all `DifferentiableLearner`s.
+        for other in self.others:
+            other.build()
+
+        # Ensure 'others' have a module reference.
+        for other in self.others:
+            if not other.module:
+                other.module = self.module
+
+    @OverrideToImplementCustomLogic
+    @override(Learner)
+    def update(
+        self,
+        batch: Optional[MultiAgentBatch] = None,
+        batches: Optional[List[MultiAgentBatch]] = None,
+        batch_refs: Optional[List[ray.ObjectRef]] = None,
+        episodes: Optional[List[EpisodeType]] = None,
+        episodes_refs: Optional[List[ray.ObjectRef]] = None,
+        data_iterators: Optional[List[ray.data.DataIterator]] = None,
```

Comment:
Happy to already NOT support these helper args anymore and go all-in on `TrainingData`. At least for this new class here.

Reply:
Happy about this, too!